File: tests/aws/services/lambda_/functions/lambda_s3_integration.py
import json
import os
import tempfile

import boto3

s3 = boto3.client("s3", endpoint_url=os.environ.get("AWS_ENDPOINT_URL"))
BUCKET_NAME = os.environ["S3_BUCKET_NAME"]


def handler(event, context):
    s3_key = context.aws_request_id
    file_size_bytes = event.get("file_size_bytes")
    if file_size_bytes is not None:
        # Upload random file if file_size_bytes is present in the event
        with tempfile.SpooledTemporaryFile() as tmpfile:
            tmpfile.write(os.urandom(file_size_bytes))
            s3.upload_fileobj(tmpfile, BUCKET_NAME, s3_key)
    else:
        # Upload the event otherwise
        s3.put_object(Bucket=BUCKET_NAME, Key=s3_key, Body=json.dumps(event))

    function_version = os.environ["AWS_LAMBDA_FUNCTION_VERSION"]
    return {"s3_key": s3_key, "function_version": function_version}

File: tests/aws/services/lambda_/functions/lambda_s3_integration.mjs
import { getSignedUrl } from "@aws-sdk/s3-request-presigner";
import { S3Client, PutObjectCommand } from "@aws-sdk/client-s3";

export const handler = async (event, context) => {
    const BUCKET_NAME = process.env.AWS_LAMBDA_FUNCTION_NAME;
    let s3;
    if (process.env.AWS_ENDPOINT_URL) {
        const CREDENTIALS = {
            secretAccessKey: 'test',
            accessKeyId: 'test',
        };

        s3 = new S3Client({
            endpoint: "http://s3.localhost.localstack.cloud:4566",
            region: 'us-east-1',
            credentials: CREDENTIALS,
        });
    } else {
        s3 = new S3Client()
    }

    const url = await getSignedUrl(
        s3,
        new PutObjectCommand({
            Bucket: BUCKET_NAME,
            Key: 'key.png',
            ContentType: 'image/png'
        }),
        {
            expiresIn: 86400
        }
    );
    return {
        statusCode: 200,
        body: JSON.stringify(url)
    };
}
